### PR TITLE
Improve Hazelast Chart to support Scaling Up/Down

### PR DESCRIPTION
**Pull request by leszko**
#### What this PR does / why we need it:
- Update Hazelcast to 3.11
- Change Docker image to `hazelcast/hazelcast` (`hazelcast/hazelcast-kubernetes` is no longer supported)
- Add Graceful Shutdown feature (used in scaling down and rolling upgrade)
- Add `resolve-not-ready-addresses` property to check also not ready Hazelcast PODs
- Increase default parameters for livenessProbe and readinessProbe (helps in stability in case of rolling upgrade and huge amount of data)
- Simplify `hazelcast.xml` configuration

#### Which issue this PR fixes
None.

#### Special notes for your reviewer:
None.

#### Checklist
- [ x ] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [ x ] Chart Version bumped
- [ x ] Variables are documented in the README.md

### CODE REVIEW

File: stable/hazelcast/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 name: hazelcast
-version: 1.0.1
-appVersion: "3.10.2"
+version: 1.0.2
```
Comment (davidkarlsen) [inline stable/hazelcast/Chart.yaml]:
```suggestion
version: 1.1.0
```

since there a bit more changes going on

**Review by davidkarlsen — CHANGES_REQUESTED** [review]

**Comment by leszko** [comment]
@davidkarlsen Thanks for the review. I applied the change suggested by you. PTAL.

**Comment by leszko** [comment]
@davidkarlsen , could you please have a look?

**Comment by leszko** [comment]
Do you think this PR is ready to merge?

**Comment by cpanato** [comment]
ping @davidkarlsen

**Comment by leszko** [comment]
Any news on that? Could we finally get it merged? It's just a small change!

**Comment by leszko** [comment]
Guys, it's already 18 days! Could you get it merged?

**Comment by leszko** [comment]
Guys, again, can we get it merged?

@davidkarlsen @foxish @lachie83 @paulczar @cpanato

**Review by davidkarlsen — APPROVED** [review]

**Pull request merged**